typed/clj.checker/src/typed/clj/checker/parse_unparse.clj

```diff
@@ -1320,21 +1320,24 @@
 
 (defn clj-primitives-fn [opts]
   (let [RClass-of @(RClass-of-var)]
-    {'byte (RClass-of 'byte opts)
-     'short (RClass-of 'short opts)
-     'int (RClass-of 'int opts)
-     'long (RClass-of 'long opts)
-     'float (RClass-of 'float opts)
-     'double (RClass-of 'double opts)
-     'boolean (RClass-of 'boolean opts)
-     'char (RClass-of 'char opts)
-     'void r/-nil}))
+    (fn [sym]
+      (case sym
+        byte (RClass-of 'byte opts)
+        short (RClass-of 'short opts)
+        int (RClass-of 'int opts)
+        long (RClass-of 'long opts)
+        float (RClass-of 'float opts)
+        double (RClass-of 'double opts)
+        boolean (RClass-of 'boolean opts)
+        char (RClass-of 'char opts)
+        void r/-nil
+        nil))))
```

Comment:
This makes a lot of sense. This map must have been created an enormous number of times.